Modules/@babylonjs/react-native/EngineHook.ts

```diff
@@ -34,33 +65,21 @@ class DOMException {
             return originalInitializeSessionAsync.apply(this, args);
         }
 
-        const cameraPermission = Platform.select({
-            android: PERMISSIONS.ANDROID.CAMERA,
-            ios: PERMISSIONS.IOS.CAMERA,
-        });
+        await requestCameraPermissionAsync();
 
-        // Only Android, iOS and Windows are supported.
-        if (cameraPermission === undefined) {
-            throw new DOMException(DOMError.NotSupportedError);
-        }
+        return originalInitializeSessionAsync.apply(this, args);
+    }
+}
 
-        // If the permission has not been granted yet, but also not been blocked, then request permission.
-        let permissionStatus = await check(cameraPermission);
-        if (permissionStatus == "denied")
-        {
-            permissionStatus = await request(cameraPermission);
-        }
+// Override the VideoTexture.CreateFromWebCamAsync to insert a camera permissions request. It would be cleaner to do this directly in the NativeCamera implementation, but there are a couple problems with that:
+// 1. React Native does not provide a way to hook into the permissions request result (at least on Android).
+// 2. If it is done on the native side, then we need one implementation per platform.
+{
+    const originalCreateFromWebCamAsync: (...args: any[]) => Promise<any> = VideoTexture.CreateFromWebCamAsync;
+    VideoTexture.CreateFromWebCamAsync = async function (...args: any[]): Promise<any> {
```

Comment:
I know this is copying the existing pattern, but I think we could make these much more strongly typed and at the same time simplify it:
```suggestion
    const originalCreateFromWebCamAsync = VideoTexture.CreateFromWebCamAsync;
    VideoTexture.CreateFromWebCamAsync = async function (...args: Parameters<typeof originalCreateFromWebCamAsync>): ReturnType<typeof originalCreateFromWebCamAsync> {
```

Reply:
I like it. I updated both method overrides to use the same type safe declarations.